src/wallet/wallet.cpp

```diff
@@ -1890,6 +1890,9 @@ void CWallet::ReacceptWalletTransactions()
         CValidationState state;
         wtx.AcceptToMemoryPool(*locked_chain, state);
     }
+
+    // Update wallet transactions with current mempool transactions.
+    chain().notifyMempoolTransactions(*this);
```

Comment:
I don't like this going in `ReacceptWalletTransactions()` for a couple of reasons:

- The name `ReacceptWalletTransactions()` suggests to me that we're reaccepting wallet transactions back into the mempool (which is what the function currently does). Adding functionality to fetch mempool txs back into the wallet changes the definition of this function so that it no longer matches the name.
- `ReacceptWalletTransactions()` is called by various import* RPCs, which don't need to refetch the mempool txs into the wallet.

I suggest moving this call into `postInitProcessing()` which is only called when a wallet is loaded.

Reply:
Fixed.

Reply:
> I suggest moving this call into `postInitProcessing()` which is only called when a wallet is loaded.

This was attempted before but there was a locking order problem, and was for that reason that I've placed `notifyMempoolTransactions` here. I've included 0994e3121 to address that.

The important thing to ensure is https://github.com/bitcoin/bitcoin/pull/15652#discussion_r268783330.